File: ovvium-backend/src/main/java/com/ovvium/services/web/controller/bff/v1/transfer/request/order/CreateOrderRequest.java
package com.ovvium.services.web.controller.bff.v1.transfer.request.order;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import static java.util.Collections.emptyList;

@Data
@NoArgsConstructor
@Accessors(chain = true)
public class CreateOrderRequest {

	private UUID userId;
	private UUID productId;
	private String serviceTime;
	private String notes;
	private List<OrderGroupChoiceRequest> groupChoices = emptyList();
	private List<UUID> selectedOptions = emptyList();

	public Optional<String> getNotes() {
		return Optional.ofNullable(notes);
	}

	public Optional<UUID> getUserId() {
		return Optional.ofNullable(userId);
	}

	public Optional<String> getServiceTime() {
		return Optional.ofNullable(serviceTime);
	}
}

File: ovvium-pos/src/app/model/request/CreateOrderRequest.ts
import { Order } from "../Order";
import { ServiceTime } from "../enum/ServiceTime";
import { OrderGroupChoiceRequest } from "./OrderGroupChoiceRequest";

export class CreateOrderRequest {

    userId?: string;
    productId: string;
    notes?: string;
    groupChoices?: OrderGroupChoiceRequest[];
    serviceTime?: ServiceTime;
    
    constructor(orderRequest: CreateOrderRequest = {} as CreateOrderRequest) {
        this.userId = orderRequest.userId;
        this.productId = orderRequest.productId;
        this.notes = orderRequest.notes;
        this.serviceTime = orderRequest.serviceTime;
        this.groupChoices = orderRequest.groupChoices;
    }

    static from (order: Order) {
        return {
            userId: order.user?.id,
            productId: order.product.id,
            notes: order.notes,
            serviceTime: order.serviceTime,
            groupChoices: order.groupChoices ? order.groupChoices.map(c => new OrderGroupChoiceRequest(c.product.id, c.notes)) : undefined
        } as CreateOrderRequest
    }
}